### ISSUE TITLE
`function-redefined` false positive with `match`

### ISSUE THREAD

**Issue by nickdrozd** [pylint-dev/pylint#8871]
### Bug description

If a function is defined differntly in `if`/`else` branches, Pylint correctly does not raise `function-redefined`:

```python
x = 1

if x == 1:
    def f() -> int:
        return 4
else:
    def f() -> int:  # no problem
        return 5
```

But in a `match` block, the warning is incorrectly raised:

```python
y = 1

match y:
    case 1:
        def g() -> int:
            return 4
    case _:
        def g() -> int:  # raises function-redefined
            return 5
```

### Command used

```shell
pylint asdf.py
```

### Pylint output

```shell
************* Module asdf
asdf.py:17:8: E0102: function already defined line 14 (function-redefined)
```

### Expected behavior

The warning should not be raised. Mypy doesn't give any errors.

### Pylint version

```shell
pylint 3.0.0b1
astroid 3.0.0a9-dev0
Python 3.11.0
```

**Comment by nickdrozd** [comment]
Whoops, duplicate of https://github.com/pylint-dev/pylint/issues/7470
